app/assets/javascripts/discourse/app/initializers/subscribe-user-notifications.js

```diff
@@ -73,17 +73,16 @@ export default {
             );
 
             if (staleIndex === -1) {
-              // this gets a bit tricky, unread pms are bumped to front
-              let insertPosition = 0;
-              if (lastNotification.notification_type !== 6) {
-                insertPosition = oldNotifications.findIndex(
-                  (n) => n.notification_type !== 6 || n.read
-                );
-                insertPosition =
-                  insertPosition === -1
-                    ? oldNotifications.length - 1
-                    : insertPosition;
+              let insertPosition = oldNotifications.findIndex(
```

Comment:
So what happens when an unread PMs shows up?

Reply:
PM notifications are high priority which means that it will be inserted at the top.

Reply:
So why did we need the “hack” to ensure unread PMs would go to the top?

Reply:
I could well just be that we added the hack on the client-side, then later we fixed server-side the mark unread PMs as high priority and forgot to remove the hack on the client-side? But I just want us to be 100% sure about that before removing that behavior on the client-side.

Reply:
We used to have only a single 'high priority' notification and that was unread PMs. Now we have bookmark reminders too.